packages/pubsub/src/batch.js

```diff
@@ -0,0 +1,220 @@
+/*!
+ * Copyright 2014 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*!
+ * @module pubsub/batch
+ */
+
+'use strict';
+
+var prop = require('propprop');
+
+/**
+ * Semi-generic Batch object used to queue up multiple requests while limiting
```

Comment:
Is this Batch object publicly exposed to our end users, or is this an implementation detail?